frontend/src/features/home-details/services/detail-services.js

```diff
@@ -0,0 +1,364 @@
+  // Chart data
+  const data = [
+    {
+      type: "bar",
+      orientation: "h",
+      marker: { color: "#8e8e8e" },
+      x: chartData.map(item => item.value[activityMetric]),
+      y: chartData.map(item => item.key)
+    }
+  ];
+
+  // Add plot layout configuration
+  const layout = {
+    autosize: true,
+    height: 250,
+    xaxis: {
+      title: i18n.t(
+        `homeActivityCountsByActivityTypeChart-xAxis-${activityMetric}`
+      ),
+      showgrid: false,
+      showline: true,
+      automargin: true,
+      showticklabels: true,
+      tickfont: {
+        size: 10
+      },
+      tickwidth: 1,
+      ticklen: 8
+    },
+    yaxis: {
+      title: i18n.t("homeActivityCountsByActivityTypeChart-yAxis-title"),
+      showgrid: false,
+      showline: true,
+      automargin: true,
+      showticklabels: true,
+      tickfont: {
+        size: 10
+      },
+      tickwidth: 1,
+      ticklen: 8
+    },
+    margin: {
+      autoexpand: true,
+      r: 10,
+      t: 60,
+      b: 20,
+      l: 80
+    },
+    bargap: 0.05,
+    showlegend: false
+  };
+
+  // Get client locale
+  const locale = i18n.locale;
+
+  // Render plot
+  Plotly.newPlot("homeActivityCountsByActivityTypeChart", data, layout, {
+    locale
+  });
+};
+
+export const getHomeActivitiesFacilitatorRoleMetrics = async (
+  homeId,
+  period
+) => {
+  try {
+    const response = await $axios.post(
+      "/methods/getHomeActivitiesFacilitatorRoleMetrics",
+      {
+        homeId,
+        period
+      }
+    );
+    return response.data;
+  } catch (error) {
+    errorNotifier(error);
+    return [];
+  }
+};
+
+export const renderFacilitatorChart = (activityMetric, chartData = []) => {
+  // Chart data
+  const data = [
+    {
+      type: "bar",
+      orientation: "h",
+      marker: { color: "#8e8e8e" },
+      x: chartData.map(item => item.value[activityMetric]),
+      y: chartData.map(item => item.key)
+    }
+  ];
+
+  // Add plot layout configuration
+  const layout = {
+    autosize: true,
+    height: 250,
+    xaxis: {
+      title: i18n.t(
+        `homeActivityCountsByFacilitatorRoleChart-xAxis-${activityMetric}`
+      ),
+      showgrid: false,
+      showline: true,
+      automargin: true,
+      showticklabels: true,
+      tickfont: {
+        size: 10
+      },
+      tickwidth: 1,
+      ticklen: 8
+    },
+    yaxis: {
+      title: i18n.t("homeActivityCountsByFacilitatorRoleChart-yAxis-title"),
+      showgrid: false,
+      showline: true,
+      automargin: true,
+      showticklabels: true,
+      tickfont: {
+        size: 10
+      },
+      tickwidth: 1,
+      ticklen: 8
+    },
+    margin: {
+      autoexpand: true,
+      r: 10,
+      t: 60,
+      b: 20,
+      l: 80
+    },
+    bargap: 0.05,
+    showlegend: false
+  };
+
+  // Get client locale
+  const locale = i18n.locale;
+
+  // Render plot
+  Plotly.newPlot("homeActivityCountsByFacilitatorRoleChart", data, layout, {
+    locale
+  });
+};
```

Comment:
Just for clarity of organization, could we please move the chart rendering functions into a different module? E.g. `features/home-details/charts/charts.js`? That way, the current file would deal *only* with data fetching, while the `charts.js` file would deal only with chart rendering. Kind of like separating forks and knives.